services/repository/mongodb/service.go

```diff
@@ -0,0 +1,204 @@
+// Package mongodb implements MongoDB repository services
+package mongodb
+
+import (
+	"context"
+
+	"github.com/decentralized-cloud/tenant/models"
+	"github.com/decentralized-cloud/tenant/services/configuration"
+	"github.com/decentralized-cloud/tenant/services/repository"
+	commonErrors "github.com/micro-business/go-core/system/errors"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/bson/primitive"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+)
+
+type mongodbRepositoryService struct {
+	configurationService configuration.ConfigurationContract
+	collectionName       string
+}
+
+// NewMongodbRepositoryService creates new instance of the RepositoryService, setting up all dependencies and returns the instance
+// Returns the new service or error if something goes wrong
+func NewMongodbRepositoryService(
+	configurationService configuration.ConfigurationContract) (repository.RepositoryContract, error) {
+
+	if configurationService == nil {
+		return nil, commonErrors.NewArgumentNilError("configurationService", "configurationService is required")
+	}
+
+	return &mongodbRepositoryService{
+		configurationService: configurationService,
+		collectionName:       "tenants",
```

Comment:
this does not have to be in the service context, it defines the collection name and will be same for all instances, can be defined simply as a const variable on the top of the file